pkg/didcomm/messenger/messenger.go

```diff
@@ -104,8 +104,6 @@ func (m *Messenger) saveMetadata(msg service.DIDCommMsgMap) error {
 		return fmt.Errorf("threadID: %w", err)
 	}
 
-	delete(msg, jsonMetadata)
```

Comment:
why was this removed?

Reply:
I have added it on the marshal level https://github.com/hyperledger/aries-framework-go/pull/1949/files#diff-fc8f058cfffcbd9722de36d08af83891R60